internal/extensions/git_registry.go

```diff
@@ -47,11 +48,34 @@ func newGitRegistry(url string, globalStore *global.GlobalStore) Registry {
 }
 
 func (g *gitRegistry) fetchGitRepository(ctx context.Context) error {
-	panic("unimplemented")
+	registry := getRegistryPath(g.globalStore)
+
+	cmd := exec.CommandContext(ctx, "git", "pull")
```

Comment:
Can we reuse some of the existing git functionality here?

Reply:
Extracted it